File: client/android/UseCase/Shipment/Detail/LoadShipmentDetailUseCase.kt
package app.mobiling.client.usecase.shipment.detail

import app.mobiling.client.contract.shipment.detail.ShipmentDetailPayload
import app.mobiling.client.data.shipment.tracking.ShipmentTrackingGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class LoadShipmentDetailUseCase(
    private val shipmentTrackingGateway: ShipmentTrackingGateway,
) {
    suspend operator fun invoke(shipmentId: String): ShipmentDetailPayload =
        shipmentTrackingGateway.loadShipmentDetail(shipmentId)
}

File: client/ios/UseCase/Shipment/Detail/LoadShipmentDetailUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct LoadShipmentDetailUseCase {
    let shipmentTrackingGateway: ShipmentTrackingGateway

    func callAsFunction(shipmentId: String) async throws -> ShipmentDetailPayload {
        try await shipmentTrackingGateway.loadShipmentDetail(shipmentId: shipmentId)
    }
}
